File: 200. Number of Islands.cpp
class Solution {
public:
    int count = 0;
    void traversal(int i,int j,vector<vector<char>>&grid,vector<vector<bool>> &visited){
        if (i >= grid.size() || j >= grid[0].size() || i < 0 || j < 0 || visited[i][j] || grid[i][j] == '0') {
            return;
        }
        visited[i][j] = true;
        traversal(i+1,j,grid,visited);
        traversal(i-1,j,grid,visited);
        traversal(i,j+1,grid,visited);
        traversal(i,j-1,grid,visited);
    }
    int numIslands(vector<vector<char>>& grid) {
        vector<vector<bool>> visited(grid.size(),vector<bool>(grid[0].size(),false));
        for(int i = 0; i<grid.size(); i++){
            for(int j = 0; j<grid[0].size(); j++){
                if(grid[i][j] == '1' && !visited[i][j]){
                    traversal(i,j,grid,visited);
                    count++;
                }
            }
        }
        return count;
    }
};

File: 200. Number of Islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        def traversal(i,j):
            if (i >= len(grid) or j >= len(grid[0]) or i < 0 or j < 0 or visited[i][j] or grid[i][j] == '0'):
                return
            visited[i][j] = True
            traversal(i-1,j)
            traversal(i+1,j)
            traversal(i,j+1)
            traversal(i,j-1)
        visited = [[False for _ in range(len(grid[0]))] for _ in range(len(grid))]
        count = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == '1' and not visited[i][j]:
                    traversal(i,j)
                    count+=1
        return count
